packages/library/src/commonMain/kotlin/io/realm/internal/RealmResultsImpl.kt

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal
+
+import io.realm.Callback
+import io.realm.Cancellable
+import io.realm.RealmObject
+import io.realm.RealmResults
+import io.realm.VersionId
+import io.realm.interop.Link
+import io.realm.interop.NativePointer
+import io.realm.interop.RealmInterop
+import kotlinx.coroutines.flow.Flow
+import kotlin.reflect.KClass
+
+// FIXME API-QUERY Final query design is tracked in https://github.com/realm/realm-kotlin/issues/84
+//  - Lazy API makes it harded to debug
+//  - Postponing execution to actually accessing the elements also prevents query parser errors to
+//    be raised. Maybe we can get an option to prevalidate queries in the C-API?
+
+class RealmResultsImpl<T : RealmObject> : AbstractList<T>, RealmResults<T> {
```

Comment:
```suggestion
internal class RealmResultsImpl<T : RealmObject> : AbstractList<T>, RealmResults<T> {
```

Reply:
Interlization is done in some of the other PRs